.github/workflows/nix.yml

```diff
@@ -0,0 +1,42 @@
+---
+name: "nix-build"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+
+  lint:
```

Comment:
Is there cost savings with having this run in a separate job? I understand the logic that the results will be the same no matter what system it is run on, so why not just run it once. But considering how fast the linting process is and how long it takes to start a job and have all the required dependencies pulled down, it might be faster to just run it `n` times.

Reply:
On second thought, the advantage of having a separate job is when there is a failure you only see it in one place. So now I'm torn between having to wait for a job I know will pass and having that clarity when it might not.